Userland/Libraries/LibCore/System.cpp

```diff
@@ -574,7 +576,11 @@ ErrorOr<DeprecatedString> gethostname()
 
 ErrorOr<void> sethostname(StringView hostname)
 {
+#if defined(AK_OS_SOLARIS)
+    int rc = ::sethostname(const_cast<char*>(hostname.characters_without_null_termination()), hostname.length());
```

Comment:
don't tell me solaris cant handle a `const char*`... please...

Reply:
Sure Solaris supports const char*,but for the sethostname function it expects a normal char*,not const,as you can verify here: https://docs.oracle.com/cd/E88353_01/html/E37843/sethostname-3c.html
You can compare that with the BSDs,which expect a const char*,so Solaris needs to be handled differently.

Reply:
thats not optimal... lets assume that indiana does not modify that buffer - otherwise, this will break the `hostname`.

Reply:
> thats not optimal... lets assume that indiana does not modify that buffer - otherwise, this will break the `hostname`.

This assumption is specified in the linked documentation, so your concern is not relevant here.